src/python/pants/binaries/binary_tool.py

```diff
@@ -5,23 +5,30 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import os
+
 from pants.binaries.binary_util import BinaryUtil
 from pants.subsystem.subsystem import Subsystem
-from pants.util.memo import memoized_method
+from pants.util.memo import memoized_method, memoized_property
 
 
 class BinaryToolBase(Subsystem):
   """Base class for subsytems that configure binary tools.
 
-  Typically, a specific subclass is created via create_binary_tool_subsystem_cls() below.
-  That subclass can be further subclassed, manually, e.g., to add any extra options.
+  Subclasses can be further subclassed, manually, e.g., to add any extra options.
 
   :API: public
   """
+  # All subsystems inheriting from BinaryToolBase will go under
+  # bin/`support_subdir` unless they specify an explicit `support_dir`.
```

Comment:
Seriously, please, let's get rid of this unnecessarily confusing concept. Or at least demote it as I suggested, via a classmethod. If someone does want to override it, we may as well let them override the `bin` part too - for all we know that may be why they need to.

Reply:
OK I see that this is possible via setting `support_dir`, but this is just confusing. Let's kill both `support_dir` and `support_subdir` entirely. We can achieve the same amount of flexibility with a classmethod that uses the value of `get_name()` by default, and can be overridden for custom logic. This is too declarative...

Reply:
Ok, I wasn't paying enough attention before, and I didn't realize your comments applied to this as well. Sorry about that. That makes sense to me now, and I've implemented it and changed the method's name to `get_support_dir()`.

Reply:
I really need to refresh pages -- I can implement the suggestion to use `get_name()` and allow overrides right now.

Reply:
Sorry for making so many commits in a row -- 83ffdda should be what we were trying to do.